### PR TITLE
Logging feature

### LINKED ISSUE #2: Implement initial README
### Dependencies

No dependencies

### Feature Request

Implement initial README with

- Project Title
- Link to GSoC page
- Goal of this project

Get started and Contribute section will be implemented in the later PR

### LINKED ISSUE #61: Enable to launch monitoring engine before the initial connection.
### Feature Request

To collect telemetry data before the initial connection to the controller, we need to introduce the feature to launch the monitoring engine on launch.

See https://docs.google.com/document/d/1gjRamLWz3NwenVifC5pYyBMmxsUjl9MjspZF0mRYeaI/edit#heading=h.8yamam8qmvkb for datails

### PR DESCRIPTION
**Pull request by Aki-7**
## What I did are
- Remove tracing feature
- Introduce logging feature
  - only logs committed via `java.util.logging`.
    - TODO: collect logs printed to standard error
  - OpenTelemetry logging implementation in Java is still a work in progress. 
    See https://github.com/open-telemetry/opentelemetry-java/pull/3415.
    So I implement some functions needed on my own.
    I'll replace the code below when OpenTelemetry Java SDK fully implements these functions.
      - BatchLogProcessor(Builder)
      - LogSinkSdkProvider(Builder)
      - OtlpGrpcLogExporter(Builder)
      - TaggedLogRecord
      - LogAdapter
      - related interfaces (LogExporter and LogProcessor)
- Stop using thread for monitoring engine
- Package monitoring engine as Java agent.
- Write user documentation including getting started (README.adoc)
- Prepare docker-compose.yaml to set up
   - OpenTelemetry collector
   - Loki for log aggregation
   - Prometheus for metric backend
   - Grafana for log and metric visualization
     - detasource is preconfigured.

## Checklist

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] All follow-ups are documented as issues and-or TODO comments
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
- [x] Please update user documentation if needed
- [x] Please update developer documentation if needed

close #61
close #2

### CODE REVIEW

File: monitoring-engine/src/main/java/io/jenkins/plugins/remotingopentelemetry/engine/log/BatchLogProcessor.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
Comment (Aki-7) [inline monitoring-engine/src/main/java/io/jenkins/plugins/remotingopentelemetry/engine/log/BatchLogProcessor.java:4]: Most of this class is just a copy of the `io.opentelemetry.sdk.logging.export.BatchLogProcessor`

File: monitoring-engine/src/main/java/io/opentelemetry/exporter/otlp/internal/LogAdapter.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package io.opentelemetry.exporter.otlp.internal;
```
Comment (Aki-7) [inline monitoring-engine/src/main/java/io/opentelemetry/exporter/otlp/internal/LogAdapter.java:1]: I understand it's bad manners. I'm sure I'll remove this class when this [PR](https://github.com/open-telemetry/opentelemetry-java/pull/3415) is merged.

File: README.adoc [hunk 3]
```diff
@@ -1,28 +1,149 @@
...
 
 image::https://img.shields.io/badge/chat-on%20slack-yellow?logo=slack[link="https://cdeliveryfdn.slack.com/archives/C023E83AMAL"]
 
+toc::[]
+
 [#introduction]
 == Introduction
 
-Collect the monitoring data of Jenkins Remoting through OpenTelemetry.
+=== Goal
```
Comment (oleg-nenashev) [inline README.adoc:14]: Could it be a separate PR?

**Review by oleg-nenashev — APPROVED** [review]
IMHO it includes many changes that should have been ideally separate PRs. The demo on Tuesday looked really good, so I would definitely recommend merging it if it is ready for alpha and evaluation by users
